test/include/psen_scan_v2/async_barrier.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2020 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#ifndef PSEN_SCAN_V2_ASYNC_BARRIER_H
+#define PSEN_SCAN_V2_ASYNC_BARRIER_H
+
+#include <future>
+#include <thread>
+#include <chrono>
+
+namespace psen_scan_v2_test
+{
+class Barrier
```

Comment:
This class provides the same mechanism as `AsyncTest` from pilz_testutils. Do you think we could replace `AsyncTest` by this class in the future? Now we have both classes used in this package, which I do not like very much.

Reply:
I agree, however, for me it is just a first step. If the new Barrier class proves itself, I also suggest to replace the old `AsyncTest` class. For now I would keep it that way. Or what do you suggest?